### ISSUE TITLE
onBeforeLoad is not called when overwriting cy.visit

### ISSUE THREAD

**Issue by bahmutov** [cypress-io/cypress#2196]
Demo: https://github.com/cypress-io/cypress-test-tiny/tree/on-before-load-when-cy-visit-overwritten

I have overwritten `cy.visit` command, passing both `onBeforeLoad` and `onLoad` - and `onBeforeLoad` callback is NOT called.

## Normal test

```js
// calls cy.visit with arguments
describe('page', () => {
  it('works', () => {
    cy.visit('https://example.cypress.io/', {
      onBeforeLoad: (win) => {
        console.log('onBeforeLoad')
      },
      onLoad: (win) => {
        console.log('onLoad')
      }
    })
  })
})
```

See how it shows both messages? both callbacks were called.

<img width="1280" alt="screen shot 2018-07-23 at 4 39 19 pm" src="https://user-images.githubusercontent.com/2212006/43083628-413c6198-8e97-11e8-954c-6dec77066e48.png">

## overwrite cy.visit

If we overwrite `cy.visit` and call the original function with callbacks, only `onLoad` is called

```js
// commands.js
Cypress.Commands.overwrite("visit", (originalVisit, url, options) => {
  console.log('calling cy.visit')
  const opts = {
    onBeforeLoad: (win) => {
      console.log('onBeforeLoad')
    },
    onLoad: (win) => {
      console.log('onLoad')
    }
  }
  return originalVisit(url, opts)
})
```

but the `onBeforeLoad` is not executed

<img width="1280" alt="screen shot 2018-07-23 at 4 38 41 pm" src="https://user-images.githubusercontent.com/2212006/43083762-89779630-8e97-11e8-8567-8d82e1b4c7cf.png">

**Comment by henrikq-vipps** [comment]
I have observed the same. A workaround is:
```
Cypress.Commands.overwrite("visit", (originalVisit, url, options) => {
  console.log('calling cy.visit')
  options.onBeforeLoad = (win) => {
      console.log('onBeforeLoad')
    }
  options.onLoad = (win) => {
      console.log('onLoad')
    }
  return originalVisit(url, options)
})
```
but you have to provide a options object, i.e.
```
    cy.visit('https://example.cypress.io/', {})
```
, and providing a default options, i.e.
```
Cypress.Commands.overwrite("visit", (originalVisit, url, options = {}) => {
```
does not allow you to omit a `options` when you `visit`.


My guess is that the original argument provided when evoking `cy.visit` in the spec is provided, and the 2. argument provided when evoking originalVisit is ignored. So, the only way to modify options is to provide a options object at `cy.visit` and mutate it in the override. 

Not a good experience to spend 2 hours figuring this out...

**Comment by akornato** [comment]
Still not fixed as of `cypress` `3.4.0`

**Comment by jennifer-shehane** [comment]
Hey @akornato, this issue was resolved and released as part of 3.1.1. If you're experiencing the same or similar issue, please open a new issue with a fully reproducible example. Thanks!
